tests/test_satfunc.py

```diff
@@ -433,6 +461,158 @@ def test_defaulted_values():
     np.testing.assert_allclose(dframe["KRW"], [0, 1])
 
 
+@pytest.mark.parametrize(
+    "dframe, expected",
+    [
+        (
+            pd.DataFrame([{"SW": 0, "KR": 0, "SATNUM": 1}]),
+            pd.DataFrame([{"SW": 0, "KR": 0, "SATNUM": 1}]),
+        ),
+        (
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.5, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.5, "KR": 0.5, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+        ),
+        (
+            # nan by missing in initalization dict:
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": 0.1, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+        ),
+        (
+            # Explicit numpy nan:
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": np.nan, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": 0.1, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+        ),
+        (
+            # python float nan:
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": float("nan"), "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": 0.1, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+        ),
+        (
+            # It will not care about column names:
```

Comment:
I think these are good, but personally I have started using `pytest.param` and `id=` which I think is great for readability when using `parameterize`:
```suggestion
            pytest.param(pd.DataFrame(
                [
                    {"SW": 0, "FOO": 0, "SATNUM": 1},
                    {"SW": 0.1, "SATNUM": 1},
                    {"SW": 1, "FOO": 1, "SATNUM": 1},
                ],
            pd.DataFrame(
                [
                    {"SW": 0, "FOO": 0, "SATNUM": 1},
                    {"SW": 0.1, "FOO": 0.1, "SATNUM": 1},
                    {"SW": 1, "FOO": 1, "SATNUM": 1},
                ]
       id="Will not care about column names",
        ),
```
I think the syntax is valid, but did not check it.

Reply:
Agree, this looks good. Will fix.